platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/server/DaemonProcessState.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.daemon.server;
+
+import org.gradle.internal.agents.AgentStatus;
+import org.gradle.internal.buildprocess.BuildProcessState;
+import org.gradle.internal.classpath.ClassPath;
+import org.gradle.internal.logging.LoggingManagerInternal;
+import org.gradle.internal.nativeintegration.services.NativeServices;
+import org.gradle.internal.service.ServiceRegistry;
+import org.gradle.internal.service.ServiceRegistryBuilder;
+import org.gradle.launcher.daemon.configuration.DaemonServerConfiguration;
+import org.gradle.launcher.daemon.registry.DaemonRegistryServices;
+
+import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Encapsulates the state of the daemon process.
+ */
+public class DaemonProcessState implements Closeable {
```

Comment:
Use a type to manage the daemon process state. There's still some logic in the daemon's main()` method that could move into this type, but for this PR, simply wrap the daemon's `BuildProcessState` and service lifecycling logic.